projects/driver_controls/src/fsm/pedal_fsm.c

```diff
@@ -1,10 +1,20 @@
-// Some events can be raised despite the FSM being in the corresponding state (such as
-// INPUT_EVENT_PEDAL_BRAKE being called while the pedal FSM is in the brake state). Even though
-// these events will not cause state transitions, they still may have data that needs to be passed
-// to CAN. As a result, these repeated events are dealt with by transitioning back into their own
-// state
+// Updates to the drive output module are driven by the update requested events
+// The throttle module should continuously provide updates to move between states, but its data
+// will be polled on state outputs. This is so that we can handle mechanical braking and cruise
+// control properly. Since the throttle module is still providing updates, state changes will cause
+// drive output updates to ensure that major changes are reflected in the periodic updates.
+
+// The overall operation of the car is governed by the FSMs. They have an implicit hierarchy:
+// * Power FSM only allows power/mechanical braking when off
```

Comment:
Can you structure this as:
```
<FSM_NAME>
- Effect on other FSMs in state <X>...
- Effect on other FSMs state <Y>...
```
It is a little hard to parse a few of them and being explicit is just nicer when it is as deterministic as an FSM.

Reply:
Yeah, I'll probably move these into a readme for driver controls now that the behavior is a little easier to understand.